sys/newlib_syscalls_default/syscalls.c

```diff
@@ -149,7 +149,16 @@ static const struct heap heaps[NUM_HEAPS] = {
  */
 void _init(void)
 {
-    /* nothing to do here */
+    if (IS_USED(MODULE_FMT)) {
+        /* dark magic optimization: A call to `setvbuf()` adds 264 B on
+         * the samr21-xpro. Instead, we manually set the stdout file to
+         * unbuffered output. We do rely on newlib internals here, but this
+         * seems to be worth it. */
+        stdout->_flags |= __SNBF;
+        stdout->_w = 0;
+        stdout->_bf._base = stdout->_p = stdout->_nbuf;
+        stdout->_bf._size = 1;
+    }
```

Comment:
Maybe now we should introduce `newlib_stdout_buffered` that handles the buffering outside libc for stdouts that benefit from that 🙈 (like `picolibc_stdout_buffered`)